lib/frontend_helpers/form_builder_helpers/switch_field.rb

```diff
@@ -3,14 +3,19 @@
 module FrontendHelpers
   module FormBuilderHelpers
     module SwitchField
-      def switch_field_group(method, options = {}, checked_value = '1', unchecked_value = '0')
+      def switch_field_group(method, options = {},
+                             checked_value = '1', unchecked_value = '0', &block)
         control_class = options.delete(:control_class)
         label_class = options.delete(:label_class)
-        label = options.delete(:label)
+        label = if block_given?
+                  @template.capture(&block)
+                else
+                  tag.p(options.delete(:label), class: label_class)
```

Comment:
aqui deberiamos de usar un `<label>` con el attributo `for` que haga match con el ID del `input` para que cuando le des click al label tambien cambie el switch.

https://www.w3schools.com/tags/tag_label.asp
<img width="671" alt="Screen Shot 2021-11-24 at 15 15 02" src="https://user-images.githubusercontent.com/17049/143324361-e3871033-4696-413a-9795-8b172903dd5b.png">

Reply:
De hecho si lo usamos, para crear un switch se le da formato al label y se esconde el checkbox. Entonces al dar click en realidad estás dando click al label.
El label que aparece a la derecha es la descripción del switch, por lo tanto  no tiene ese funcionamiento, en ningún lugar al dar click a la descripción se hace el toggle del switch